sfge/src/test/java/com/salesforce/graph/visitor/PathScopeVisitorTest.java

```diff
@@ -110,4 +112,36 @@ public void testParameterNameShadowsField() {
         MatcherAssert.assertThat(
                 TestUtil.apexValueToString(value.getValues().get(1)), equalTo("value2"));
     }
+
+    @Test
+    public void testVariableNameReuseThrowsUserActionException() {
+        String sourceCode =
+                "public class MyClass {\n"
+                        + "   public static void doSomething() {\n"
+                        + "       String myStr = 'hi';\n"
+                        + "       String myStr = 'hello';\n"
+                        + "   }\n"
+                        + "}\n";
+
+        UserActionException thrown =
+                assertThrows(
+                        UserActionException.class,
+                        () -> TestRunner.walkPath(g, sourceCode),
+                        "UserActionException should've been thrown before this point");
+
+        MatcherAssert.assertThat(thrown.getMessage(), containsString("MyClass:4"));
+    }
+
+    @Test
+    public void testParameterNameAndFieldDoNotClash() {
```

Comment:
@jfeingold35 Added a test for the case you mentioned and we seem to be doing fine.